### PR TITLE
cp: 'cp foo foo' with backup should not fail

### PR DESCRIPTION
**Pull request by sylvestre**
Should fix tests/cp/backup-1.sh

### CODE REVIEW

**Comment by sylvestre** [comment]
`Warning: Congrats! The gnu test tests/cp/backup-1 is no longer failing!`

**Comment by tertsdiepraam** [comment]
Code looks good, but the test seems to be a bit flaky on mac and windows.

File: tests/by-util/test_cp.rs [hunk 1]
```diff
@@ -1626,3 +1626,26 @@ fn test_cp_overriding_arguments() {
...
+    let at = &scene.fixtures;
+    at.touch("source.txt");
+    scene
+        .ucmd()
+        .arg("--backup=simple")
+        .arg("--suffix=.b")
+        .arg("source.txt")
+        .arg("source.txt")
+        .succeeds();
+    assert!(at.file_exists("source.txt.b"));
```
Comment (jfinkels) [inline tests/by-util/test_cp.rs:1642]: Hmm on GNU cp v8.30, I see the following:
```sh
$ touch source.txt && cp --backup=simple --suffix=.b source.txt source.txt
cp: 'source.txt' and 'source.txt' are the same file
```
The command exits with status 1 and `source.txt.b` does not get created. Am I missing something here? I notice that the GNU test case in `tests/cp/backup-1.sh` has a `--force` flag that is not present here.

Comment (sylvestre) [reply tests/by-util/test_cp.rs:1642]: i had the --force but it was failing on Windows (not implemented). 
I will put it back

File: tests/by-util/test_cp.rs [hunk 2]
```diff
@@ -1626,3 +1626,26 @@ fn test_cp_overriding_arguments() {
...
 }
+
+#[test]
+fn test_cp_same_file_backup() {
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    at.touch("source.txt");
+    scene
+        .ucmd()
+        .arg("--backup=simple")
```
Comment (sylvestre) [inline tests/by-util/test_cp.rs:1637]: Add force back

```suggestion
        .arg("--force")
        .arg("--backup=simple")
```

**Review by jfinkels — APPROVED** [review]
Makes sense to me

**Comment by tertsdiepraam** [comment]
What was the status on this one? This merge conflict seems to imply that it has already been fixed, right? Both versions have the `options.backup == BackupMode::NoBackup` check.
```
<<<<<<< cp-same
fn handle_existing_dest(source: &Path, dest: &Path, options: &Options) -> CopyResult<()> {
    if paths_refer_to_same_file(source, dest)? && options.backup == BackupMode::NoBackup {
        return Err(format!("{}: same file", context_for(source, dest)).into());
=======
/// Decide whether source and destination files are the same and
/// copying is forbidden.
///
/// Copying to the same file is only allowed if both `--backup` and
/// `--force` are specified and the file is a regular file.
fn is_forbidden_copy_to_same_file(
    source: &Path,
    dest: &Path,
    options: &Options,
    source_in_command_line: bool,
) -> bool {
    // TODO To match the behavior of GNU cp, we also need to check
    // that the file is a regular file.
    let dereference_to_compare =
        options.dereference(source_in_command_line) || !source.is_symlink();
    paths_refer_to_same_file(source, dest, dereference_to_compare)
        && !(options.force() && options.backup != BackupMode::NoBackup)
}
```

**Comment by jfinkels** [comment]
I think it can be closed as already fixed because of pull request #3966.

**Pull request closed without merging**